File: Haskell/somSeq.hs
module Main(main)where

main = do putStrLn "Digite uma sequência de números (um por linha)"
          putStrLn "Para terminar digite o valor zero"
          soma <- lerESomar
          putStr "A soma dos números digitados é "
          print soma

lerESomar = do n <- readLn
               if n==0
                   then return 0
                   else do somaResto <- lerESomar
                           return (n+somaResto)

File: C/somSeq.c
#include<stdio.h>
int main(){
	int soma=0;
	int x;
	printf("Digite uma sequência de números (um por linha)\n");
    printf("Para terminar digite o valor zero\n");
    for(;;){
    	scanf("%d",&x);
    	if(x==0){
    		break;
    	}
    	soma+=x;
    }
    printf("A soma dos números digitados e' %d\n",soma);
	return 0;
}
